src/main/scala/Models.scala

```diff
@@ -1,21 +1,38 @@
+import spray.json._
+import DefaultJsonProtocol._
+import sangria.execution.deferred.HasId
+import sangria.schema._
+import sangria.macros.derive._
+
 case class Category(
   id: Int,
-  name: String
-) {
-  def styles(allStyles: List[Style]): List[Style] =
-    allStyles.filter(_.category_id == id)
+  name: String)
+
+object Category {
+  implicit val categoryHasId = HasId[Category, Int](_.id)
+  implicit val categoryFormat = jsonFormat2(Category.apply)
+
+  implicit val graphqlType: ObjectType[Repository, Category] = deriveObjectType(
+    ObjectTypeDescription("A category"),
+    AddFields(Field("styles", ListType(Style.graphqlType),
+      resolve = c => c.ctx.styleFetcher.deferRelSeq(c.ctx.styleByCategory, c.value.id))))
 }
```

Comment:
I was trying to keep my models GraphQL-free, separating layers sort of, so that case classes contain the business logic and the schema file focuses on the GraphQL integration. Was that a stupid idea?